### DISCUSSION TITLE
Is is safe to keep using useId hook while IdProvider is deprecated?

### DISCUSSION THREAD

**Discussion by denik1981** [radix-ui/primitives#1229] [Help]
I have read the source code for the `@radix-ui/rect-id` **useId** hook and it doesn't seem to use the deprecated IdProvider as its context. The codebase I'm collaborating with uses **useId** to generate default unique ids for non-radix components AND IdProvider as well. I believe I can safely remove IdProvider and keep using **useId** until React@18 useId hook is merged (stable) into the project.
What do you think?

**Comment by benoitgrelard** [comment]
That's correct, it is safe.
